scripts/web/install.sh

```
@@ -71,21 +71,42 @@ setup_install_dir() {
     mkdir -p "$INSTALL_DIR"
 }
 
+# Detect user's shell type
+detect_shell_type() {
+    # Use the SHELL environment variable which contains user's default shell
+    local shell_path="${SHELL:-/bin/bash}"
+    local shell_name="$(basename "$shell_path")"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)